### ISSUE TITLE
bootstrap 5 + badge/icon

### ISSUE THREAD

**Issue by drma-tech** [Megabit/Blazorise#3162]
it doesn't seem to be getting the font colors correctly, according to the context.

bootstrap 4
![image](https://user-images.githubusercontent.com/12803050/143323954-bc7584c7-1efd-4356-ab1a-1a45f7e9ebb5.png)

bootstrap 5
![image](https://user-images.githubusercontent.com/12803050/143323828-b350eaef-59e2-4628-9b58-b853ac78a1fb.png)

**Comment by stsrki** [comment]
Do you assign any `TextColor` to the component?

**Comment by Jimmys20** [comment]
Hello, it's because in Bootstrap 5 the contextual badge classes were removed and now you have to use background and text utility classes to style badges: https://getbootstrap.com/docs/5.0/migration/#badges. So instead of `<span class="badge-warning"></span>` you have to use `<span class="bg-warning text-dark"></span>`.

**Comment by drma-dev** [comment]
yes, basically before (bootstrap 4) it automatically detected according to the context.

**Comment by stsrki** [comment]
I didn't notice this before because I always have Blazorise theming ON, and it properly shows the color. Without it, it fallback to BS5 colors.

You can either enable theming or manually define `TextColor` parameter.

**Comment by drma-dev** [comment]
[https://blazorise.com/docs/theming](https://blazorise.com/docs/theming)

are you talking about this documentation? As I understand it, you define fixed colors or a way to change colors at runtime.

but you can do this for example: when the badge is dark, the font is light, and when the badge is light, the font is dark. because that's how bootstrap 4 worked.

**Comment by drma-dev** [comment]
I think I found what's missing. I went to look at the bootstrap page:
![image](https://user-images.githubusercontent.com/12803050/143785972-ec045fe9-1c23-455f-afb3-1bb358878bb5.png)
this css class is not read by the component.
![image](https://user-images.githubusercontent.com/12803050/143785989-06a8e3c4-d102-46ce-8e0a-e1bfc3a89793.png)

**Comment by ulvesked** [comment]
In bootstrap 5.2.0, this has been added as a generic helper. 

https://getbootstrap.com/docs/5.2/helpers/color-background/

It should be refactored to be set as `text-bg-warning` instead of just `bg-warning`

**Comment by stsrki** [comment]
> In bootstrap 5.2.0, this has been added as a generic helper.
> 
> https://getbootstrap.com/docs/5.2/helpers/color-background/
> 
> It should be refactored to be set as `text-bg-warning` instead of just `bg-warning`

That's a nice catch. Thanks for letting us know. I have tested and the fix is easy enough to go into the next 1.7 patch.
